### PR TITLE
added IF collateral support proposal

### PR DESCRIPTION
**Pull request by shibe-dev**
Proposal to add impossible finance's governance token (IF) as collateral

### CODE REVIEW

**Comment by calchulus** [comment]
Happy to answer any questions you guys might have!

File: impossible-finance-collateral-support.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Rationale
+
+Whitelisting IF allows us to mint IFUSD synthetic tokens for use in our ecosystem. This design enables synthetic stablecoins tied to a defi ecosystem's health. This enables defi teams to create a treasury with better risk management long term.
+
+We chose 200 IF as the fee because this is around 450-500 USD which historically is close to what other collateral tokens have used. We are happy to adjust accordingly. We would like to confirm if the fee is a "one-time" fee per user, per minter, or per synthetic deployer?
+
+# Implementation
+
+This change has no implementation other than the two aforementioned governor transactions that will be proposed.
```
Comment (therealhexi) [inline impossible-finance-collateral-support.md:37]: `governor transactions` didn't read well. Do you mean `governance transaction`?

File: impossible-finance-collateral-support.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# Security Considerations
+
+The primary usage of IF tokens will be to create value-backed IFUSD synthetics for use in our ecosystem. We note that IF tokens is a volatile currency with a low trading volume currently. Its price is also very closely tied to the health of the Impossible Finance ecosystem. As such, users that are considering using IF as collateral should ensure that their positions are sufficiently collaterized.
+
+Another inherent risk in this system is the anyswap.exchange/multichain.xyz bridges which in theory, have access to mint an infinite amount of tokens on chains non-native to the asset. The bridge designs uses secure Multiparty-Computation (MPC) technology which the Impossible team has reviewed along with the audit report of the implementation. Our findings are that the technology and implementation is secure and sufficiently decentralized and should pose negligible risk.
```
Comment (therealhexi) [inline impossible-finance-collateral-support.md:43]: Would be great to link the audit report and MPC so future readers can cross-verify :-)

**Review by therealhexi — COMMENTED** [review]
Initial review

File: impossible-finance-collateral-support.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+By approving this UMIP, Impossible Finance will be able to mint synthetic ifUSD stablecoins tied to the IF ecosystem, backed by Impossible Finance (IF) tokens as collateral. This design will create a pool of stablecoins tied to our ecosystem (and the IF token) which allows us to manage risk and not be reliant on price movements of other stablecoins.
+
+# Technical Specifications
+
+1. The [IF address](https://etherscan.io/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code), 0xb0e1fc65c1a741b4662b813eb787d369b8614af1, will need to be whitelisted to add it as collateral. This whitelist was introduced in UMIP-8.
+
+We note that the current version of IF is a native [BSC asset](https://bscscan.com/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code) and we're working with anyswap.exchange and Andre's multichain.xyz bridges to bridge our governance token to ETH. It will be also the mechanism by which we migrate from the old STAX token to the new IF token across all the chains we want to participate in.
```
[the remark this answers is not in the document]
Comment (calchulus) [reply impossible-finance-collateral-support.md:25]: we will keep total supply at 1:1, and retire the old STAX token. Total eventual supply 21M

File: impossible-finance-collateral-support.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+We chose 200 IF as the fee because this is around 450-500 USD which historically is close to what other collateral tokens have used. We are happy to adjust accordingly. We would like to confirm if the fee is a "one-time" fee per user, per minter, or per synthetic deployer?
+
+# Implementation
+
+This change has no implementation other than the two aforementioned governor transactions that will be proposed.
+
+# Security Considerations
+
+The primary usage of IF tokens will be to create value-backed IFUSD synthetics for use in our ecosystem. We note that IF tokens is a volatile currency with a low trading volume currently. Its price is also very closely tied to the health of the Impossible Finance ecosystem. As such, users that are considering using IF as collateral should ensure that their positions are sufficiently collaterized.
```
[the remark this answers is not in the document]
Comment (calchulus) [reply impossible-finance-collateral-support.md:41]: STAX was a fair launch on BSC and all info can be seen here:
https://bscscan.com/token/0x0da6ed8b13214ff28e9ca979dd37439e8a88f6c4 

IF will carry the same user holding distributions

File: impossible-finance-collateral-support.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+We note that the current version of IF is a native [BSC asset](https://bscscan.com/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code) and we're working with anyswap.exchange and Andre's multichain.xyz bridges to bridge our governance token to ETH. It will be also the mechanism by which we migrate from the old STAX token to the new IF token across all the chains we want to participate in.
+
+2. A final fee of 200 IF needs to be enabled in the store contract.
+
+# Rationale
+
+Whitelisting IF allows us to mint IFUSD synthetic tokens for use in our ecosystem. This design enables synthetic stablecoins tied to a defi ecosystem's health. This enables defi teams to create a treasury with better risk management long term.
+
+We chose 200 IF as the fee because this is around 450-500 USD which historically is close to what other collateral tokens have used. We are happy to adjust accordingly. We would like to confirm if the fee is a "one-time" fee per user, per minter, or per synthetic deployer?
```
[the remark this answers is not in the document]
Comment (calchulus) [reply impossible-finance-collateral-support.md:33]: got it, thanks!

File: impossible-finance-collateral-support.md (hunk 2 above)
Comment (calchulus) [reply impossible-finance-collateral-support.md:43]: https://github.com/anyswap/Anyswap-Audit/blob/master/SlowMist/AnySwap%20CrossChain-Bridge%20Security%20Audit%20Report.pdf

File: impossible-finance-collateral-support.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+| Created             | April 27th, 2021                                              |
+[Discourse Link](https://discourse.umaproject.org/t/adding-impossible-finance-if-as-collateral/1017)
+
+# Summary
+
+This UMIP will add Impossible Finance tokens (IF) as a collateral currency for UMA. This process involves adding our governance token to the whitelist an adding a flat
+
+# Motivation
+
+Impossible Finance's IF token will be the core asset of the Impossible Swap and savings products within the multichain world. It acts as the fee accrual asset within the swap ecosystem. Adding onto this, Impossible Finance's next suite of products focuses on offering a fair fundraising and a decentralised incubator/launchpad ecosystem. We believe this will be a core feature to drive consistant volume in the future on our swap.
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply impossible-finance-collateral-support.md:17]: The token has 0 transfers, 0 holders and a total supply of 0 it seems.

File: impossible-finance-collateral-support.md (hunk 2 above)
Comment (smb2796) [reply impossible-finance-collateral-support.md:43]: Awesome! Can you add this into the UMIP?

File: impossible-finance-collateral-support.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (shibe-dev) [reply impossible-finance-collateral-support.md:17]: correct - we've just deployed the contract allowing a 1:1 transfer of our original governance token, STAX. We have not started processing the migration but will be doing once multichain is live

File: impossible-finance-collateral-support.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+By approving this UMIP, Impossible Finance will be able to mint synthetic ifUSD stablecoins tied to the IF ecosystem, backed by Impossible Finance (IF) tokens as collateral. This design will create a pool of stablecoins tied to our ecosystem (and the IF token) which allows us to manage risk and not be reliant on price movements of other stablecoins.
+
+# Technical Specifications
+
+1. The [IF address](https://etherscan.io/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code), 0xb0e1fc65c1a741b4662b813eb787d369b8614af1, will need to be whitelisted to add it as collateral. This whitelist was introduced in UMIP-8.
+
+We note that the current version of IF is a native [BSC asset](https://bscscan.com/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code) and we're working with anyswap.exchange and Andre's multichain.xyz bridges to bridge our governance token to ETH. It will be also the mechanism by which we migrate from the old STAX token to the new IF token across all the chains we want to participate in.
+
+2. A final fee of 200 IF needs to be enabled in the store contract.
```
[the remark this answers is not in the document]
Comment (shibe-dev) [reply impossible-finance-collateral-support.md:27]: the current version of the governance token is STAX is here: https://coinmarketcap.com/currencies/stablexswap/

[the remark this answers is not in the document]
Comment (shibe-dev) [reply impossible-finance-collateral-support.md:27]: we're doing a 1:1 migration to IF tokens, the IF contract on BSC is here: https://bscscan.com/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code. This token will be multichain-linked to ETH IF tokens which is at this address: https://etherscan.io/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code

File: impossible-finance-collateral-support.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+We note that the current version of IF is a native [BSC asset](https://bscscan.com/address/0xb0e1fc65c1a741b4662b813eb787d369b8614af1#code) and we're working with anyswap.exchange and Andre's multichain.xyz bridges to bridge our governance token to ETH. The model for bridging is mint/burn. It will be also the mechanism by which we migrate from the old STAX token to the new IF token across all the chains we want to participate in. This migration is a 1:1 token migration, IF + stax tokens have the same circulating supply
+
+2. A final fee of 200 IF needs to be enabled in the store contract.
+
+# Rationale
+
+Whitelisting IF allows us to mint IFUSD synthetic tokens for use in our ecosystem. This design enables synthetic stablecoins tied to a defi ecosystem's health. This enables defi teams to create a treasury with better risk management long term.
+
+We chose 200 IF as the fee because this is around 450-500 USD which historically is close to what other collateral tokens have used. We are happy to adjust accordingly.
```
Comment (Reinis-FRP) [inline impossible-finance-collateral-support.md:33]: I understand STAX->IF will be one-way migration on BSC. If it is not possible to un-migrate, there would be no arbitrage mechanism keeping IF token price pegged to STAX price, hence, it is not possible to estimate the value of proposed final fee before the migration and establishment of IF token market price.

File: impossible-finance-collateral-support.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+This change has no implementation other than the two aforementioned governance transactions that will be proposed.
+
+# Security Considerations
+
+The primary usage of IF tokens will be to create value-backed IFUSD synthetics for use in our ecosystem. We note that IF tokens is a volatile currency with a low trading volume currently. Its price is also very closely tied to the health of the Impossible Finance ecosystem. As such, users that are considering using IF as collateral should ensure that their positions are sufficiently collaterized.
+
+Here's a link for STAX token implementation: https://bscscan.com/token/0x0da6ed8b13214ff28e9ca979dd37439e8a88f6c4. STAX was a fair launch on BSC.
+
+Another inherent risk in this system is the anyswap.exchange/multichain.xyz bridges which in theory, have access to mint an infinite amount of tokens on chains non-native to the asset. The bridge designs uses secure Multiparty-Computation (MPC) technology which the Impossible team has reviewed along with the audit report of the implementation. Our findings are that the technology and implementation is secure and sufficiently decentralized and should pose negligible risk.
```
Comment (Reinis-FRP) [inline impossible-finance-collateral-support.md:45]: Just to clarify, who is controlling mpc/vault addresses on BSC and Ethereum networks for IF token contracts?

**Pull request closed without merging**